src/store/ChartAdapterStore.ts

```diff
@@ -237,38 +241,58 @@ export default class ChartAdapterStore {
-            } else if (e.type === 'touchmove') {
+            const { pageX, pageY } = e.changedTouches[0];
+
+            if (['touchmove', 'touchend'].includes(e.type)) {
                 const nonScrollableAreaWidth = chartNode.offsetWidth - this.mainStore.chart.yAxisWidth;
                 const { left } = chartNode.getBoundingClientRect();
 
                 if (this.touchValues.x && this.touchValues.y) {
-                    const deltaX = Math.abs(screenX - this.touchValues.x);
-                    const deltaY = Math.abs(screenY - this.touchValues.y);
+                    const deltaX = Math.abs(pageX - this.touchValues.x);
+                    const deltaY = Math.abs(pageY - this.touchValues.y);
+                    this.touchValues.deltaYTotal = (this.touchValues.deltaYTotal ?? 0) + (this.touchValues.y - pageY);
                     const isVerticalScroll = deltaY > deltaX;
                     const x = pageX - left;
-                    if (x < nonScrollableAreaWidth && isVerticalScroll && !this.scrollChartParentOnTouchTimer) {
-                        this.touchValues.yOnTouchEnd = undefined;
-                        this.scrollChartParentOnTouchTimer = setTimeout(() => {
-                            this.scrollableChartParent?.scrollBy({
-                                top: screenY - Number(this.touchValues.yOnTouchEnd ?? this.touchValues.y),
+                    const isForcedScrollArea = x < nonScrollableAreaWidth;
+                    const shouldForceMaxScroll =
+                        Math.abs(Number(this.touchValues.deltaYTotal)) > 10 && e.type === 'touchend';
+                    if (isForcedScrollArea) {
+                        if (shouldForceMaxScroll) {
+                            clearTimeout(this.scrollChartParentOnTouchTimer);
+                            this.scrollableChartParent?.scrollTo({
+                                top:
+                                    Number(this.touchValues.deltaYTotal) < 0
+                                        ? 0
+                                        : this.scrollableChartParent.scrollHeight,
                                 behavior: 'smooth',
                             });
                             this.scrollChartParentOnTouchTimer = undefined;
-                        }, 300);
+                            this.touchValues = { ...this.touchValues, deltaYTotal: 0 };
+                        } else if (isVerticalScroll && !this.scrollChartParentOnTouchTimer) {
+                            this.scrollChartParentOnTouchTimer = setTimeout(() => {
+                                this.scrollableChartParent?.scrollBy({
+                                    top: this.touchValues.deltaYTotal,
+                                    behavior: 'smooth',
+                                });
+                                this.scrollChartParentOnTouchTimer = undefined;
```

Comment:
Inside of timeout timer can be cleared only by undefined assignment or this will work too:⤵️? clearTimeout(this.scrollChartParentOnTouchTimer);

Reply:
we don't want to clear the timer inside it because it has already been executed, it won't be needed.
the purpose of `this.scrollChartParentOnTouchTimer = undefined;` is to ensure that we initiate the next `setTimeout` execution only after the current one has finished execution.